pkg/internal/nodeinfo/cpu_models.go

```diff
@@ -0,0 +1,333 @@
+	cpuModelCores := make(map[string]*resource.Quantity)
+	cpuModelMemory := make(map[string]*resource.Quantity)
+
+	// Calculate cluster totals for normalization
+	totalNodes := len(nodes)
+	var totalCpu float64
+	var totalMemory float64
+	for _, node := range nodes {
+		if cpu := node.Status.Capacity.Cpu(); cpu != nil {
+			totalCpu += cpu.AsApproximateFloat64()
+		}
+		if mem := node.Status.Capacity.Memory(); mem != nil {
+			totalMemory += mem.AsApproximateFloat64() / (1024 * 1024 * 1024) // Convert to GB
+		}
+	}
+
+	for _, node := range nodes {
+		cpuModels := extractCpuModels(node.Labels)
+		memoryCapacity := node.Status.Capacity.Memory()
+		cpuCapacity := node.Status.Capacity.Cpu()
+
+		for _, cpuModel := range cpuModels {
+			cpuModelCounts[cpuModel]++
+
+			if cpuCapacity != nil {
+				if cpuModelCores[cpuModel] == nil {
+					cpuModelCores[cpuModel] = resource.NewQuantity(0, resource.DecimalSI)
+				}
+				cpuModelCores[cpuModel].Add(*cpuCapacity)
+			}
+
+			if memoryCapacity != nil {
+				if cpuModelMemory[cpuModel] == nil {
+					cpuModelMemory[cpuModel] = resource.NewQuantity(0, resource.BinarySI)
+				}
+				cpuModelMemory[cpuModel].Add(*memoryCapacity)
+			}
+		}
+	}
+
+	newHash := generateHash(cpuModelCounts, cpuModelCores, cpuModelMemory)
+
+	cpuModelInfo.lock.Lock()
+	defer cpuModelInfo.lock.Unlock()
+
+	if cpuModelInfo.lastUpdateHash == newHash {
+		return false
+	}
+
+	cpuModelInfo.totalNodes = totalNodes
+	cpuModelInfo.totalCpu = totalCpu
+	cpuModelInfo.totalMemory = totalMemory
+	cpuModelInfo.models = make(map[string]*v1beta1.CpuModelInfo)
+	for cpuModel, count := range cpuModelCounts {
+		cpuStr := "0"
+		if cpuModelCores[cpuModel] != nil {
+			cpuStr = cpuModelCores[cpuModel].String()
+		}
+
+		memStr := "0"
+		if cpuModelMemory[cpuModel] != nil {
+			memStr = cpuModelMemory[cpuModel].String()
+		}
+
+		cpuModelInfo.models[cpuModel] = &v1beta1.CpuModelInfo{
+			Name:      cpuModel,
+			Benchmark: getCpuModelPassMarkScore(cpuModel),
+			Nodes:     count,
+			Cpu:       cpuStr,
+			Memory:    memStr,
+		}
+	}
+
+	cpuModelInfo.lastUpdateHash = newHash
+	return true
+}
+
+func extractCpuModels(labels map[string]string) []string {
+	cpuModels := sets.New[string]()
+
+	for key, value := range labels {
+		if strings.HasPrefix(key, CpuModelLabelPrefix) {
+			// Extract CPU model name from the label key
+			cpuModel := strings.TrimPrefix(key, CpuModelLabelPrefix)
+			// Only include if the value is "true" (meaning the node supports this model)
+			if value == "true" {
+				cpuModels.Insert(cpuModel)
+			}
+		}
+	}
+
+	return cpuModels.UnsortedList()
+}
+
+func getCpuModelPassMarkScore(cpuModel string) int {
+	// Try exact match first
+	if score, exists := cpuModelPassMarkScores[cpuModel]; exists {
+		return score
+	}
+
+	// Try to find a partial match for models with suffixes
+	for modelName, score := range cpuModelPassMarkScores {
+		if strings.Contains(cpuModel, modelName) || strings.HasPrefix(cpuModel, modelName) {
```

Comment:
if cpuModek has a prefix of modelName, it also contains it, right?


```suggestion
		if strings.Contains(cpuModel, modelName) {
```